### ISSUE TITLE
Automatic elimination of unwanted title extensions like "Remastered..."

### ISSUE THREAD

**Issue by hans-juergen** [david-sabata/web-scrobbler#880]
Hi, when listening to the new Beatles catalogue on [Spotify](https://open.spotify.com/artist/3WrFJ7ztbogyGnTHbHJFl2) and [Deezer](http://www.deezer.com/artist/1), I noticed that the native scrobbler of Deezer's web player does no longer submit the unwanted title extensions like "Remastered 2009" etc. to Last.fm. They seem to have implemented an internal filter for that which is a very welcome feature of course, because Spotify's title extensions spoil your Last.fm statistics due to automatically creating a new page for every "new" track.

Long story short: is it possible to implement such an automatic filter in your web scrobbler, too?

**Comment by nlavergne** [comment]
Couldn't agree more, came here for this exact reason.
Please implement this great suggestion!!

**Comment by janbaykara** [comment]
Would be good. The trouble in the implementation is in recognising a 'title extension' like "Remastered 2009" from a genuine track name like "2009" or "Remastered" or "The Remastered" or "In 2009" or something. Which is not to say it's impossible - we'd probably just need to build a nice big sample list to test filtering against.

**Comment by hans-juergen** [comment]
Most of the time they are in brackets or use another hyphen, so that would make it easier. By the way, Last.fm is currently doing this work internally for an upcoming catalogue correction which will be released in some weeks, but having it in the scrobbler extension would of course support this effort.

**Comment by david-sabata** [comment]
Good place for such feature will probably be a pipeline stage, working with filters (reg.exps maybe?) that were set up somewhere in settings.

But if you say that L.FM already tries to do such thing, it's possible that they will include it in their _autocorrect_ feature. Can you go to the extension settings and tick "Use autocorrect"? This allows L.FM to automatically correct the track data if it decides to.

**Comment by hans-juergen** [comment]
As far as I know, the new catalogue will use different methods and rules for auto-correction etc., that is why the old auto-correction by suggestions and votes is not embedded on the new site anymore.

**Comment by alexesprit** [comment]
We have a new filter mechanism that could be configured to remove the gabrage from song metadata. So, it's could be good to have a list of strings that should be removed.

**Comment by hans-juergen** [comment]
To give you an example, check my library pages of Pink Floyd albums and tracks which are still a mess:

http://www.last.fm/user/hjbardenhagen/library/music/Pink+Floyd

The staff is still working on internal correction layers for these issues, but has not published them yet, and as far as I know there is no ETA for it.

**Comment by alexesprit** [comment]
I added some filters based on your Last.fm playback history in my repo: https://github.com/alexesprit/web-scrobbler/tree/deezer-filter

Could be great if you test it and tell if I missing something that should be removed.

**Comment by hans-juergen** [comment]
I guess you mean this list: https://github.com/alexesprit/web-scrobbler/commit/caeaf6670a3c8e8fd171ee6c620fb193936597eb

Looks OK for me, but I mainly scrobble to Last.fm with the Spotify desktop client, so what you see in my library comes from there most of the time. Deezer might have other methods to add info to album or track titles. One thing that Spotify got completely wrong is adding the "Remaster" extension even twice sometimes, see e.g. http://www.last.fm/user/hjbardenhagen/library/music/Pink+Floyd/The+Wall+%5B2011+-+Remaster%5D+(2011+Remastered+Version) for the album The Wall, and another time for every track on it.

**Comment by alexesprit** [comment]
So, should this filter be applied to Spotify too?

> adding the "Remaster" extension even twice sometimes

This filter should remove all duplicated "Remaster" garbage in this case.

**Comment by hans-juergen** [comment]
I would say yes, but see janbaykara's comment from January above. It might happen that sometimes "Remaster" is a legit part of a track or album title or even of an artist name.

Spotify also changes their titles now and then, so it is possible that this double "Remaster" extension does not exist in their catalogue anymore. But it probably won't hurt to have it in your filter list nevertheless.

**Comment by alexesprit** [comment]
Ok, I got it.

The removal rules shouldn't affect on the song metadata in cases described by janbaykara.

**Comment by hans-juergen** [comment]
Good. I just checked the Spotify catalogue for The Wall again, and they still call the album this way:

https://open.spotify.com/album/1ITpsfAcoaSe4i1tUXm44D

Of course this is an extended version of The Wall with demo tracks on a third CD, so it should not be confused with the original version. So that is one reason why it is tricky to correct these things, see e.g. the related release group on MusicBrainz and how many different releases there are:

https://musicbrainz.org/release-group/f2026101-945b-3d05-9ef4-aa718fc3feef
